src/psyclone/parse/kernel.py

```diff
@@ -611,6 +624,29 @@ def get_kernel_metadata(name, ast):
         raise ParseError("Kernel type {0} does not exist".format(name))
     return ktype
 
+def get_kernel_interface(ast):
+    '''Takes the kernel module parse tree and returns the interface part
+    of the parse tree.
+
+    :param ast: parse tree of the kernel module code
+    :type ast: :py:class:`fparser.one.block_statements.BeginSource`
+
+    :returns: Name 'name' of the parse tree of the interface 
+    :rtype: :str `fparser.one.block_statements.Interface.name`
+
+    :returns 'None' if there is no interface, the interface has no name or has no module procedures
+    '''
+    iname = None
```

Comment:
I have to admit that I got to this point and then got very confused about what is going on. Please could you tidy up (for pycodestyle) and add comments. I suspect you could extend your new `get_kernel_interface` routine to return the list of kernel names that the interface defines and that might avoid doing this second walk here?